### ISSUE TITLE
[bug] gnu triplet for os=Linux arch=x86 is suboptimal

### ISSUE THREAD

**Issue by t-8ch** [conan-io/conan#10211]
### Environment Details (include every applicable attribute)
  * Operating System+version: ArchLinux current
  * Compiler+version: gcc 11.1.0
  * Conan version: Current git head ( 1.24.0-1085-g106e54c9 )
  * Python version: Python 3.10.1

### Description

The GNU triplet that is created for arch=x86 on Linux is "x86-linux-gnu".
(This is only for Linux, other operating systems are translated as "i686" instead of "x86")

This causes issue when cross-compiling packages that are using gnulibs [`stack-direction.m4`](https://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=blob;f=m4/stack-direction.m4;h=b33920f2885ae7aeb83e6f6ccaaf30b0e76c4e43;hb=f8eed11b15e9141d061900e4068ea1f3ba9b63f6) (like `m4` itself) because the "x86" machine is not recognized by that script.

[I talked](https://lists.gnu.org/archive/html/bug-gnulib/2021-12/msg00093.html) to the maintainer to gnulibs `stack-direction.m4` and they recommended to instead use the more correct "i686" instead of "x86".

Looking at [the source](https://github.com/conan-io/conan/blob/106e54c96118a1345899317bf3bebaf393c41574/conan/tools/gnu/get_gnu_triplet.py#L18), I'm also wondering why there is a special case for Linux.

### Steps to reproduce (Include if Applicable)

```sh
$ cat conanfile.txt 
[build_requires]
m4/1.4.19
$ conan install . -s arch=x86 --build m4
...
> source_subfolder/configure '--prefix=/home/t-8ch/.conan/data/m4/1.4.19/_/_/package/6173abd988ee2e3159da6cce836868055fdb1cb4' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share' --build=x86_64-linux-gnu --host=x86-linux-gnu

...
checking for stack direction... unknown
...
compilation errors because the stack direction was not determined
```

**Comment by matlec** [comment]
Can confirm this issue in Conan 2.0.1. The wrong triplet prevents cross-compiling m4/1.4.19 to x86 on Linux. Any updates or work-arounds?

**Comment by matlec** [comment]
Replying to myself... to work-around this issue, we can override the GNU triplet in a profile for x86:
```
[conf]
tools.gnu:host_triplet=i686-linux-gnu
```

**Comment by memsharded** [comment]
Fixed in #15699, it will be in next 2.2 release
